contracts/ServiceNodeRewards.sol

```diff
@@ -755,11 +797,40 @@ contract ServiceNodeRewards is Initializable, Ownable2StepUpgradeable, PausableU
     /// the owner.
     /// @param newMax The new maximum non-signer threshold
     function setBLSNonSignerThresholdMax(uint256 newMax) public onlyOwner {
-        require(newMax > 0, "The new BLS non-signer threshold must be non-zero");
+        if (newMax <= 0)
+            revert PositiveNumberRequirement();
         blsNonSignerThresholdMax = newMax;
         emit BLSNonSignerThresholdMaxUpdated(newMax);
     }
 
+    /// @notice Max Claim amount to use in the Exponential Moving Average check
+    /// before allowing the user to claim. If the claimed amount over 24 hours
+    /// exceeds this then the claim will fail
+    /// @param newMax The new maximum non-signer threshold
+    function setClaimEMAThreshold(uint256 newMax) public onlyOwner {
+        if (newMax <= 0)
+            revert PositiveNumberRequirement();
+        claimEMAThreshold = newMax;
+        emit ClaimEMAThresholdUpdated(newMax);
+    }
+
+    /// @notice Updates the Exponential Moving Average (EMA) with a new value
+    /// @dev This function calculates a time-weighted EMA using the provided new value and current time
+    /// @param ema The TimeWeightedEMA struct to update
+    /// @param newValue The new value to incorporate into the EMA.
+    /// @param currentTime The current timestamp, usually `block.timestamp`
+    /// @custom:formula EMA = EMA_old + α * (newValue - EMA_old)
+    /// where α = 1 - 0.5 ^ (timeDiff / halfLife)
+    function updateEMA(TimeWeightedEMA storage ema, uint256 newValue, uint256 currentTime) internal {
+        uint256 timeDiff = currentTime - ema.lastUpdateTime;
+        uint256 SCALING_FACTOR = 262144; // 2^18
+        if (timeDiff > 0) {
+            uint256 alpha = SCALING_FACTOR - (SCALING_FACTOR * SCALING_FACTOR) / (SCALING_FACTOR + (timeDiff * SCALING_FACTOR / ema.halfLife));
+            ema.value = ((SCALING_FACTOR - alpha) * ema.value / SCALING_FACTOR) + newValue;
```

Comment:
The problem here is that it means the EMA value doesn't have a long term mean that goes to the actual withdrawal rates, but rather tends perpetually too high.  e.g. if I simulate withdrawing 100k per day spaced exactly 1 day apart then $\alpha = 86400/(86400 + 36000)$, and so the long-term $s$ value tends to:
```math
\begin{align*}
s &= (1-\alpha) s + x
  &= \frac{x}{\alpha}
```
which is about 141.7k.  (In simulations I get this, or a slightly different 143.3k when I use integer math).  And so basically trying to withdraw, say, 1.6M per day for 2 days would trigger the EMA's limit.

One potential fix to this is to change ${} + x$ to ${} + \alpha x$ to fix the biased mean, but that introduces a different problem: short intervals would not powerful enough to push the EMA value upwards (because if you have $\delta T = 1$ second then $\alpha$ is tiny), and so short spans would allow huge withdrawals.

Reply:
Havent fully digested this but i made mention above that this formula in the comments is out of date. I removed the alpha from modifying the newValue after some experimentation.

the issue i was facing is that if alpha is close to 1 (aka 2 withdrawals in quick succession) then it reduces the new value down too much. So you could withdraw 1 token, then 1 second later withdraw > 2M because the alpha would reduce the new value back below the limit

Reply:
Its probably worth noting that if nobody claims for some multiple of the half life then the old value does get multiplied by a value close to zero. So it does have the capacity to reset itself.